mesonbuild/compilers/compilers.py

```diff
@@ -637,9 +637,13 @@ def __repr__(self):
     def can_compile(self, src):
         if hasattr(src, 'fname'):
             src = src.fname
-        suffix = os.path.splitext(src)[1].lower()
+        # We try first in case sensitive mode detect .C as C++ code instead of C
+        suffix = os.path.splitext(src)[1]
         if suffix and suffix[1:] in self.can_compile_suffixes:
             return True
+        # We keep the old case insensitive behaviour
+        if suffix.lower() and suffix[1:] in self.can_compile_suffixes:
```

Comment:
I think this should be `suffix.lower()[1:] in ...`. The two lower at the start of the line doesn't really do anything.

Also the duplicate check for suffix not being empty could be refactored.

Reply:
Yes here I did a mistake, I don't really understand why ```suffix and suffix[1:]```